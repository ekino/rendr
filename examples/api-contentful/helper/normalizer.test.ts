import { createNormalizer, loadJson } from "@ekino/rendr-loader-contentful";
import {
  normalizeBlockText,
  normalizeArticle,
  normalizeAuthor,
  normalizeBlockFooter,
  normalizeBlockHeader,
  normalizeBlockRawConfiguration,
} from "./normalizer";

const files = [
  "rendr_article",
  "rendr_author",
  "rendr_block_text",
  "rendr_block_raw_configuration",
];

describe("test normalization", () => {
  files.forEach((file) => {
    it(`test ${file}.json`, () => {
      const normalizer = createNormalizer({
        rendr_author: normalizeAuthor,
        rendr_article: normalizeArticle,
        rendr_block_text: normalizeBlockText,
        rendr_block_footer: normalizeBlockFooter,
        rendr_block_header: normalizeBlockHeader,
        rendr_block_raw_configuration: normalizeBlockRawConfiguration,
      });

      const entry = loadJson(
        `${__dirname}/__fixtures__/normalizer/${file}.json`
      );
      const block = normalizer(entry);

      expect(block).toBeDefined();
      expect(block).toMatchSnapshot();
    });
  });
});
