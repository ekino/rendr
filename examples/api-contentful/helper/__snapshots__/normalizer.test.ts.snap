// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test normalization test rendr_article.json 1`] = `
Object {
  "abstract": "The challenge of using a Headless CMS but still want to control how the template is managed",
  "authors": Array [
    Object {
      "biography": "Just a guy...",
      "id": "4JSAPT1jf0J8XJa0cg1aCa",
      "image": Object {
        "id": "2pSrRlUsweaxu4tW7w1uVv",
        "title": "Thomas Rabaix's Picture",
        "url": "//images.ctfassets.net/f5bx295b472q/2pSrRlUsweaxu4tW7w1uVv/dd3e01d59c68d8b07101ddf953a120f7/DSCF2267.jpg",
      },
      "jobTitle": "CTO London",
      "name": "Thomas Rabaix",
      "slug": "thomas-rabaix",
      "social": Object {
        "facebook": undefined,
        "linkedin": undefined,
        "twitter": "th0masr",
      },
    },
  ],
  "blocks": Array [
    Object {
      "container": "body",
      "id": "57UPFciwX1EEPgkZNh0gXn",
      "meta": Object {},
      "order": 99,
      "settings": Object {
        "contents": "The article contents with the markdown format! http://images.ctfassets.net/f5bx295b472q/2pSrRlUsweaxu4tW7w1uVv/dd3e01d59c68d8b07101ddf953a120f7/DSCF2267.jpg",
        "image": Object {
          "enabled": false,
          "id": "-",
          "title": "No picture",
          "url": "",
        },
        "image_position": "left",
        "mode": "standard",
        "subtitle": "",
        "title": "",
      },
      "type": "rendr.text",
    },
    Object {
      "container": "body",
      "id": "3CxKOYXVpLCPCwNCXGdn4N",
      "meta": Object {},
      "order": 99,
      "settings": Object {
        "contents": "The second part of the journey",
        "image": Object {
          "enabled": false,
          "id": "-",
          "title": "No picture",
          "url": "",
        },
        "image_position": "left",
        "mode": "standard",
        "subtitle": "",
        "title": "",
      },
      "type": "rendr.text",
    },
  ],
  "id": "4yrf1lgxv3VY8K1jpjkog0",
  "images": Object {
    "header": Object {
      "id": "5vV7exdcps9vwi2I1XEFfP",
      "title": "Header image",
      "url": "//images.ctfassets.net/f5bx295b472q/5vV7exdcps9vwi2I1XEFfP/da39b569566d118b2711f8e5ec210c76/Screenshot_2019-08-21_at_12.02.06.png",
    },
    "list": Object {
      "id": "4BRlYkVTQwQBcGb2Gft8Nb",
      "title": "Main image",
      "url": "//images.ctfassets.net/f5bx295b472q/4BRlYkVTQwQBcGb2Gft8Nb/700ef1992bd71efbe6ce42f6741a1ca2/Screenshot_2019-08-21_at_12.02.00.png",
    },
  },
  "published_at": "2019-08-15T00:00",
  "seo": Object {
    "description": "The description",
    "keywords": "The keywords",
  },
  "slug": "headless-cms-challenge",
  "title": "Headless CMS challenge",
  "type": undefined,
  "website": Object {
    "cache": Object {
      "sharedTtl": 0,
      "ttl": 0,
    },
    "countryCode": "FRA",
    "culture": "fr_FR",
    "domains": Array [
      "ekino.com",
    ],
    "enabled": true,
    "id": "2g5lXpzZ0jPPwQViziSjtK",
    "mainMenu": Object {},
    "name": "Ekino",
    "order": 99,
    "path": "/",
    "settings": Object {},
  },
}
`;

exports[`test normalization test rendr_author.json 1`] = `
Object {
  "biography": "CTO at Ekino. Sonata Project founder. Symfony Expert. Lately doing python, go, ansible and cloud platforms.",
  "id": "7pOMMe1TEojuNC9gKdxR0k",
  "image": Object {
    "enabled": false,
    "id": "-",
    "title": "No picture",
    "url": "",
  },
  "jobTitle": "CTO",
  "name": "Thomas Rabaix",
  "slug": "thomas-rabaix",
  "social": Object {
    "facebook": undefined,
    "linkedin": undefined,
    "twitter": "th0masr",
  },
}
`;

exports[`test normalization test rendr_block_raw_configuration.json 1`] = `
Object {
  "container": "body",
  "id": "",
  "meta": Object {},
  "order": 1,
  "settings": Object {},
  "type": "article.view",
}
`;

exports[`test normalization test rendr_block_text.json 1`] = `
Object {
  "container": "body",
  "id": "6NS6bLDrWzCC5I2n9NI3nj",
  "meta": Object {},
  "order": 99,
  "settings": Object {
    "contents": "Hello, here my content, with the markdown format...",
    "image": Object {
      "enabled": false,
      "id": "-",
      "title": "No picture",
      "url": "",
    },
    "image_position": "left",
    "mode": "standard",
    "subtitle": "",
    "title": "The content",
  },
  "type": "rendr.text",
}
`;
